src/proof/print_expr.h

```diff
@@ -0,0 +1,84 @@
+/*********************                                                        */
+/*! \file print_expr.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief The module for printing Lfsc proof nodes
+ **/
+
+#include "cvc5_private.h"
+
+#ifndef CVC4__PROOF__PRINT_EXPR_H
+#define CVC4__PROOF__PRINT_EXPR_H
+
+#include <iostream>
+#include <map>
+
+#include "expr/node.h"
+#include "proof/proof_node.h"
+
+namespace cvc5 {
+namespace proof {
+
+/** 
+ * A term, type or a proof. This class is used for printing combinations of
+ * proofs terms and types.
+ */
+class PExpr
+{
+ public:
+  PExpr() : d_node(), d_pnode(nullptr), d_typeNode() {}
+  PExpr(Node n) : d_node(n), d_pnode(nullptr), d_typeNode() {}
+  PExpr(const ProofNode* pn) : d_node(), d_pnode(pn), d_typeNode() {}
+  PExpr(TypeNode tn) : d_node(), d_pnode(nullptr), d_typeNode(tn) {}
+  ~PExpr() {}
+  /** The node, if this p-exression is a node */
+  Node d_node;
+  /** The proof node, if this p-expression is a proof */
+  const ProofNode* d_pnode;
+  /** The type node, if this p-expression is a type */
+  TypeNode d_typeNode;
```

Comment:
maybe use a union?

Reply:
This would require an identifier indicating Node|ProofNode|TypeNode. This isn't performance intensive code (only used in printers), so I'd prefer keeping it simpler.

Reply:
Sounds good.